### PR TITLE
Adding a promotional card

### PR DESCRIPTION
**Pull request by gjones**
### Summary
For time such as trial countdown, or perhaps to advertise a new feature or our involvement in a high profile conference. Here's a card that is designed to draw a little more attention. 

CleanShot 2023-11-29 at 15 27 09@2x

https://github.com/ClickHouse/click-ui/assets/305167/9409aa55-dce9-4aa3-9d9d-d29eca42d2c3

### CODE REVIEW

File: src/components/CardPromo/CardPromo.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  display: flex;
+  width: 100%;
+  align-items: center;
+  justify-content: flex-start;
+  cursor: pointer;
+
+  ${({ theme }) => `
+    background: ${theme.click.card.promotion.color.background.default};
+    color: ${theme.click.card.promotion.color.text.default}; 
+    border-radius: ${theme.click.card.horizontal.radii.all};
```
Comment (vineethasok) [inline src/components/CardPromo/CardPromo.tsx:31]: Should we use horizontal card here?
If we are using horizontal Card, Should we create a common wrapper for both?

File: src/components/CardPromo/CardPromo.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    padding: ${theme.click.card.horizontal.space.y} ${theme.click.card.horizontal.space.x};
+    gap: ${theme.click.card.horizontal.space.gap};
+    transition: .2s ease-in-out all;
+  
+    &:hover {
+      background: ${theme.click.card.promotion.color.background.hover};
+    }
+
+    `}
```
Comment (vineethasok) [inline src/components/CardPromo/CardPromo.tsx:40]: Could fix alignment

File: src/components/CardPromo/CardPromo.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+  ${({ theme }) => `
+    background: ${theme.click.card.promotion.color.background.default};
+    color: ${theme.click.card.promotion.color.text.default}; 
+    border-radius: ${theme.click.card.horizontal.radii.all};
+    padding: ${theme.click.card.horizontal.space.y} ${theme.click.card.horizontal.space.x};
+    gap: ${theme.click.card.horizontal.space.gap};
+    transition: .2s ease-in-out all;
+  
+    &:hover {
```
Comment (vineethasok) [inline src/components/CardPromo/CardPromo.tsx:36]: Should we have a focus or active state styling on keyboard nav?

File: src/components/CardPromo/CardPromo.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+import { Icon, IconName, Text } from "@/components";
+
+export interface CardPromoProps extends HTMLAttributes<HTMLDivElement> {
+  label: string;
+  icon: IconName;
+  dismissible?: boolean;
+}
+const Background = styled.div`
+  ${({ theme }) => `
+  background-image: ${theme.click.card.promotion.color.stroke.default};
```
Comment (vineethasok) [inline src/components/CardPromo/CardPromo.tsx:12]: Should we use the stroke for the background image?
Is this intended?
Should we rename it to something else?

File: src/components/CardPromo/CardPromo.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
+import { HTMLAttributes, useState } from "react";
+import styled from "styled-components";
+import { Icon, IconName, Text } from "@/components";
+
+export interface CardPromoProps extends HTMLAttributes<HTMLDivElement> {
+  label: string;
+  icon: IconName;
+  dismissible?: boolean;
```
Comment (vineethasok) [inline src/components/CardPromo/CardPromo.tsx:8]: Could add a iconDir to decide on left and right icon

File: src/styles/variables.light.json [hunk 6]
```diff
@@ -505,6 +505,26 @@
...
+          "icon": {
+            "default": "#161517",
+            "hover": "#161517"
+          },
+          "background": {
+            "default": "lch(97.2 1.57 272 / 0.9)",
+            "hover": "lch(91.8 1.07 266)"
+          },
+          "stroke": {
+            "default": "linear-gradient(174deg, #ABABAB 7.59%, #D4D4D4 30.01%)",
```
Comment (vineethasok) [inline src/styles/variables.light.json:524]: May be we can add it inside background and divide between default and image

**Review by vineethasok — CHANGES_REQUESTED** [review]
Need to add it in components and types to export it to the user when importing

File: src/components/CardPromo/CardPromo.tsx (hunk 4 above)
Comment (gjones) [reply src/components/CardPromo/CardPromo.tsx:12]: It's intended, turns out my bizarre design is very hard to implement in CSS as there is no linear-gradient option on borders, most workarounds result in the border radius stops working. 

So this fairly simple workaround is to have a div behind the div with a padding of 1px. This way the background image pokes out around the side acting as a gradient border. So I think the token name is correct even if the implementation is a little unorthodox.

File: src/components/CardPromo/CardPromo.tsx (hunk 2 above)
Comment (gjones) [reply src/components/CardPromo/CardPromo.tsx]: Done

File: src/components/CardPromo/CardPromo.tsx (hunk 3 above)
Comment (gjones) [reply src/components/CardPromo/CardPromo.tsx]: Yeah, I added both

**Review by vineethasok — APPROVED** [review]

**Pull request merged**